utility/src/main/java/org/oppia/android/util/namevalidator/NameValidator.kt

```diff
@@ -0,0 +1,39 @@
+package org.oppia.android.util.namevalidator
+
+class NameValidator {
+  companion object {
+    private const val MATCH_INTERNATIONAL_LETTERS_REJECT_SPACES =
+      "^.[\\w\\u00BF-\\u1FFF\\u2C00-\\uD7FF.'\\-]+\$"
+
+    private const val REJECT_SYMBOLS = "^[^#*!@\$%^&()_+=\\\\|\\]\\[\":;?/><,`~{}]*\$"
+
+    private const val REJECT_REPEATED_USE_OF_ALLOWED_SYMBOLS = "[\\.'-]{2}"
+
+    fun validate(name: String): Boolean {
+      return (
+        notEmptyNoSpacesAndContainsLetters(name) &&
+          noNumbers(name) &&
+          noSymbols(name) &&
+          noRepeatedUseOfAllowedSymbols(name)
+        )
+    }
+
+    private fun notEmptyNoSpacesAndContainsLetters(name: String): Boolean {
+      return name.matches(Regex(MATCH_INTERNATIONAL_LETTERS_REJECT_SPACES))
```

Comment:
Suggest saving these ``Regex`` objects in lazy-initialized ``val``s at the companion object level.

Reply:
i'm not sure about this one... Is there a reference in the codebase I could use?

Reply:
It would look something like:

```kotlin
private val letterRegex by lazy { Regex("...") }
```

Then you would use ``letterRegex`` when needed.